litestar/serialization.py

```diff
@@ -55,12 +54,6 @@ def _enc_constrained_date(date: ConstrainedDate) -> str:
     return date.isoformat()
 
 
-def _enc_dto(dto: DTOInterface) -> Any:
-    if isinstance(ret := dto.to_encodable_type(), bytes):
-        return msgspec.Raw(ret)
```

Comment:
I don't fully understand why this is not needed anymore. Can you explain?

Reply:
In the previous pattern:

```py
dto_type.from_data(data, connection)
```
Would create a new dto instance and assign the data to it, (`self._data = data`). So then later, when the msgspec encoder would call `dto.to_encodable_type()` the data is retrieved from the instance and converted to an encodable type, and returned.

Now:

```py
dto_type(connection).data_to_encodable_type(data)
```
The data is handed to the DTO and converted to the encodable type in one step, so it just doesn't fit that pattern any more.

Reply:
Okay that's a bit easier. Thanks for the explanation!

Reply:
Specifically for the case where the dto returns bytes, the value is given to the `Response` object which doesn't call `render()` on it if it is bytes.